### ISSUE TITLE
Update variable with value from WebSocket message: when variable = 0

### ISSUE THREAD

**Issue by UWSCIA** [bitfocus/companion-module-generic-websocket#12]
Feedback:
"Generic Websocket Label": Update variable with value from WebSocket message
JSON Path (blank if not json) = tps
variable = tps

JSON Recieved = {"status":1003,"statusDescription":"TPS Usage","tps":0,"message":"0 ticks from {1}"}

Stored variable when tps is 0 = null / empty
Stored variable when anything other than 0 is stored correctly.

Seems to be an issue when phrasing JSON INT variables when equal to 0...
Syntax for this maybe mixing a boolean result?

**Comment by krzyc** [comment]
I have tested it in 2.2.0, 2.4.2 and also current unmerged version on 3.0.0-beta and tps is set to 0. Is it still a problem?

**Comment by MonsieurNaexec** [comment]
Yep, still facing the issue, tested in 2.4.2 (2.4.2+4911-v2-4.2-fcb5a863)
It seems to happen when the variable is not initialized yet (at companion startup, when disabling/enabling the module, on server connection lost, ...)

Here's an example log:
```
23.04.17 23:23:12 instance(websocket): Message received: {"variable":0}
23.04.17 23:23:12 instance(websocket): Connection opened
23.04.17 23:23:12 instance(80ktbMCuJ): Enable instance websocket
```
Then, when inspecting the variable, the value is blank.
When another message is received and the value is not 0, the variable is set correctly.
Then, if another message contains again the value 0, it finally works and the variable is set to 0.

Maybe somewhere it checks if the value is truthy before initializing the variable?

**Comment by krzyc** [comment]
I think this is upstream behavior resulting from using != for comparison which is not type aware and treats 0, false and '' as same values (not updating). Possible solution is to initialize it with null value, which is not equal to 0.
https://github.com/bitfocus/companion/blob/cd0b68c621cd9bf0fecd0d74d8831cf164882a70/lib/variable.js#L533
@Julusian Do you know whether treating 0, false and '' variables as not updated is bug or feature?

**Comment by Julusian** [comment]
I think that is a bug. That could be fixed in the beta branch

**Comment by krzyc** [comment]
Fixed in Companion beta branch (3.0).
